use-cases/bed-allocation/src/main/java/org/acme/bedallocation/domain/BedPlan.java

```diff
@@ -50,28 +45,21 @@ public void setDepartments(List<Department> departments) {
         this.departments = departments;
     }
 
-    public void setDepartmentSpecialties(List<DepartmentSpecialty> departmentSpecialties) {
-        this.departmentSpecialties = departmentSpecialties;
-    }
-
-    public List<DepartmentSpecialty> getDepartmentSpecialties() {
-        return departmentSpecialties;
-    }
-
-    public List<Room> getRooms() {
-        return rooms;
-    }
-
-    public void setRooms(List<Room> rooms) {
-        this.rooms = rooms;
-    }
-
-    public List<Bed> getBeds() {
-        return beds;
+    @JsonIgnore
+    @ProblemFactCollectionProperty
+    public List<Room> extractRooms() {
+        return departments.stream().filter(d -> d.getRooms() != null).flatMap(d -> d.getRooms().stream()).toList();
```

Comment:
The next getter formats the stream better. Let's do the same thing, stay consistent.